src/gz/explorer.c

```diff
@@ -302,14 +302,25 @@ static int draw_proc(struct menu_item *item,
   /* render room */
   if (data->state == STATE_RENDER && data->room_file) {
     /* initialize rcp for rendering rooms */
-    static void *zbuf = NULL;
-    if (!zbuf)
-      zbuf = memalign(64, 2 * Z64_SCREEN_WIDTH * Z64_SCREEN_HEIGHT);
-    if (!zbuf)
-      return 0;
-
     static Gfx init_rcp[] = {
+      /* clear Z buffer */
+      gsSPTexture(0xFFFF, 0xFFFF, 0, G_TX_RENDERTILE, G_OFF),
+      gsSPLoadGeometryMode(G_ZBUFFER),
+      gsDPPipeSync(),
```

Comment:
This should be equivalent and marginally smaller if you want to try it out, but don't let it hold anything up.
```c
    static Gfx init_rcp[] = {
      /* rsp settings */
      gsSPSegment(0x08, &null_dl),
      gsSPSegment(0x09, &null_dl),
      gsSPSegment(0x0A, &null_dl),
      gsSPSegment(0x0B, &null_dl),
      gsSPSegment(0x0C, &null_dl),
      gsSPSegment(0x0D, &null_dl),
      gsSPTexture(0xFFFF, 0xFFFF, 0, G_TX_RENDERTILE, G_OFF),
      gsSPLoadGeometryMode(G_ZBUFFER | G_SHADE | G_CULL_BACK | G_LIGHTING |
                           G_SHADING_SMOOTH),
      /* clear Z buffer */
      gsDPPipeSync(),
      gsDPSetOtherMode(G_AD_DISABLE | G_CD_DISABLE | G_CK_NONE | G_TC_FILT |
                       G_TF_BILERP | G_TT_NONE | G_TL_TILE | G_TD_CLAMP |
                       G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
                       G_AC_NONE | G_ZS_PRIM | CVG_DST_CLAMP | FORCE_BL |
                       ZMODE_OPA | Z_UPD | IM_RD |
                       GBL_c1(G_BL_CLR_IN, G_BL_A_IN, G_BL_CLR_MEM, G_BL_1MA) |
                       GBL_c2(G_BL_CLR_IN, G_BL_A_IN, G_BL_CLR_MEM, G_BL_1MA)),
      gsDPSetCombineLERP(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
      gsDPSetPrimDepth(0x7FFF, 0),
      gsDPFillRectangle(0, 0, Z64_SCREEN_WIDTH - 1, Z64_SCREEN_HEIGHT - 1),
      /* other rdp settings */
      gsDPPipeSync(),
      gsDPSetDepthSource(G_ZS_PIXEL),
      gsDPSetTexturePersp(G_TP_PERSP),
      gsDPSetCycleType(G_CYC_2CYCLE),
      gsDPSetEnvColor(0xFF, 0xFF, 0xFF, 0xFF),
      gsDPSetFogColor(0x00, 0x00, 0x00, 0x00),
      gsDPSetScissor(G_SC_NON_INTERLACE, 32, 32,
                     Z64_SCREEN_WIDTH - 32, Z64_SCREEN_HEIGHT - 32),
      gsSPEndDisplayList(),
    };
```

Reply:
I'll test it soon, thanks. Out of curiousity:
* What restores `Z_CMP` here?
* Why is `gsDPSetPrimColor(0, 0, 0, 0, 0, 0)` not necessary?

Reply:
- Nothing restores `Z_CMP` here, the original and previous DL also did not make any guarantees about the state of `Z_CMP`. Whatever is rendered after this will set a new render mode that restores `Z_CMP`, presumably.
- Primcolor only needs to be set if the combiner uses it, and here the combiner sources all color channels from the fixed 0 input, so primcolor goes unused and doesn't need to be set.

Reply:
Thanks for explaining, I've retested on Ares and Dolphin and it seems to work